### PR TITLE
Add getting Gym detail, clean up

### PR DESCRIPTION
**Pull request by Paramon**

### CODE REVIEW

File: pogo/demo.py [hunk 1]
```diff
@@ -185,6 +185,33 @@ def sortCloseForts(session):
...
+            dist = Location.getDistance(
+                latitude,
+                longitude,
+                fort.latitude,
+                fort.longitude
+            )
+            if fort.type == 0:
+                ordered_forts.append({'distance': dist, 'fort': fort})
+
+    ordered_forts = sorted(ordered_forts, key=lambda k: k['distance'])
```
Comment (rubenvereecken) [inline pogo/demo.py]:
```
# I don't think flatten is not a function but it's probably in itertools
fort = filter(flatten(cell.forts for cell in cells.map_cells), f: f.type == 0)
sorted(forts, key=lambda f: Location.getDistance(latitude, longitude, f.latitude, f.longitude)
```

Let's keep this beauty groomed.

**Comment by dnsBlah** [comment]
Shouldn't you leave it out the way, and use only 1 function to retrieve forts.
As a Pokestop is a Fort, and a Gym as well.

Just make the difference in findClosestPokestop, findClosestGym
or just... findClosestFort if it doesnt matter where you go, and how extended the bot is ;-)

**Comment by Paramon** [comment]
Thanks. I will.

**Comment by Paramon** [comment]
If I understand correctly, we need sortCloseFort function like this:
```
def sortCloseFort(session, type = 1):
    logging.info("Sorting Nearest Forts:")
    cells = session.getMapObjects()
    latitude, longitude, _ = session.getCoordinates()
    forts = filter(lambda f: f.type == 0, [fort for cell in cells.map_cells for fort in cell.forts])
    return sorted(forts, key=lambda f: Location.getDistance(latitude, longitude, f.latitude, f.longitude))
```

And from findClosestGym call with type for Gym

File: pogo/demo.py [hunk 2]
```diff
@@ -165,32 +165,27 @@ def getInventory(session):
...
+    logging.info("Finding Nearest Gym:")
+    return sortCloseForts(session, 0)[0] if sortCloseForts(session) else None
 
 
 # Find the fort closest to user
 def findClosestFort(session):
     # Find nearest fort (pokestop)
     logging.info("Finding Nearest Fort:")
-    return sortCloseForts(session)[0]
+    return sortCloseForts(session)[0] if sortCloseForts(session) else None
```
Comment (Paramon) [inline pogo/demo.py]: We can have no fort in area
```
return sortCloseForts(session)[0]
IndexError: list index out of range
```

Comment (dmadisetti) [inline pogo/demo.py]: Notice how that would call sortCloseForts(session) twice, invoking multiple requests. Set forts to a temp variable instead.

**Comment by Paramon** [comment]
Yes. As a Pokemon select.

I suggest create something like "cell worker" class.
Who extract all data from cells. 
- Pokemons:
- Forts:
 
And move all logic of finding sorting cell data to it.

**Comment by dnsBlah** [comment]
I believe there is no 'type' given in a Gym 'fort'
I would remove the attribute in sortCloseForts

And do the loop and type check in
findClosestPokestop()
   forts = sortCloseForts
   for fort in forts:
       if fort.type and fort.type == 1:
           return fort

findClosestGym()
   forts = sortCloseForts
   for fort in forts:
       if not fort.type or fort.type != 1:
           return fort

   
Something like that :)

**Pull request closed without merging**